test/functional/test_framework/test_framework.py

```diff
@@ -476,15 +487,23 @@ def _initialize_chain(self):
             for node in self.nodes:
                 node.wait_for_rpc_connection()
 
-            # Create a 200-block-long chain; each of the 4 first nodes
+            # Create a 199-block-long chain; each of the 4 first nodes
             # gets 25 mature blocks and 25 immature.
+            # The 4th node gets only 24 immature blocks.
             for i in range(2):
                 for peer in range(4):
                     for j in range(25):
+                        if i == 1 and peer == 3 and j == 24:
+                            # Don't store the very last block in the cache to avoid having it go stale
+                            # See IsInitialBlockDownload()
```

Comment:
This needs more explanation, since iiuc it's very relevant to the bug you're trying to squash. What cache is going stale and what is the consequence of that? And in what we should one see `IsInitialBlockDownload`? What about the cache causes nodes to think they're in or out of IBD? The absolute age of the block in the cache?

It's explained a bit in the next commit, but that's in a very different place.

Reply:
There are other conditions that keep our node in IBD, but the one we care about here is the tip age (difference between now and the time the miner claimed to have mined the block)